src/components/learner-credit-management/search/CatalogSearchResults.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { useContext, useEffect, useMemo } from 'react';
+import { connectStateResults } from 'react-instantsearch-dom';
+import PropTypes from 'prop-types';
+
+import { SearchPagination, SearchContext } from '@edx/frontend-enterprise-catalog-search';
+import { FormattedMessage, injectIntl } from '@edx/frontend-platform/i18n';
+import {
+  Alert, CardView, DataTable, Skeleton, TextFilter
+} from '@edx/paragon';
+
+import CourseCard from '../cards/CourseCard';
+
+const ERROR_MESSAGE = 'An error occurred while retrieving data';
+
+const SKELETON_DATA_TESTID = 'enterprise-catalog-skeleton';
+
+export const BaseCatalogSearchResults = ({
+  error,
+  isSearchStalled,
+  paginationComponent: SearchPagination,
+  searchResults,
+  searchState,
+  setNoContent,
+}) => {
+  const courseColumns = useMemo(
+    () => [
+      {
+        Header: 'Course name',
+        accessor: 'title',
+      },
+      {
+        Header: 'Partner',
+        accessor: 'partners[0].name',
+      },
+      {
+        Header: 'A la carte course price',
+        accessor: 'first_enrollable_paid_seat_price',
+      },
+      {
+        Header: 'Associated catalogs',
+        accessor: 'enterprise_catalog_query_titles',
+      },
+    ],
+    [],
+  );
+
+  const tableData = useMemo(
+    () => searchResults?.hits || [],
+    [searchResults?.hits],
+  );
+  
+  // TODO: handle onClick
+  const renderCardComponent = (props) => <CourseCard {...props} onClick={null} />;
+  const { refinements } = useContext(SearchContext);
+
+  const page = refinements.page || (searchState ? searchState.page : 0);
+
+  useEffect(() => {
+    setNoContent(searchResults === null || searchResults?.nbHits === 0);
+  }, [searchResults, setNoContent]);
+
+  if (isSearchStalled) {
+    return (
+      <div data-testid={SKELETON_DATA_TESTID}>
+        <Skeleton className="m-1 loading-skeleton" height={25} count={5} />
+      </div>
+    );
+  }
+  if (error) {
+    return (
+      <Alert className="mt-2" variant="warning">
+        <FormattedMessage
+          id="catalogs.catalogSearchResults.error"
+          defaultMessage="{message}: {fullError}"
+          description="Error message displayed when results cannot be returned."
+          values={{ message: ERROR_MESSAGE, fullError: error.message }}
+        />
+      </Alert>
+    );
+  }
+
+  return (
+    <DataTable
+      columns={courseColumns}
+      data={tableData}
+      defaultColumnValues={{ Filter: TextFilter }}
+      initialState={{
+        pageSize: 15,
+        pageIndex: 0,
```

Comment:
[question] I'm setting the initial prop `pageSize` to display 15 results per the figma mock, but the datatable displays 20. The same behavior happens when I set the prop `pageSize` on line 98 to 15 as well. Why is this happening?